driver-core/src/main/com/mongodb/internal/connection/Authenticator.java

```diff
@@ -104,4 +105,10 @@ public void reauthenticate(final InternalConnection connection) {
         authenticate(connection, connection.getDescription());
     }
 
+    public void reauthenticateAsync(final InternalConnection connection, final SingleResultCallback<Void> callback) {
+        beginAsync().thenRun((c) -> {
+            authenticateAsync(connection, connection.getDescription(), c);
+        }).finish(callback);
```

Comment:
```suggestion
        authenticateAsync(connection, connection.getDescription(), callback);
```

Reply:
I think the convention should be to always wrap. (For example, wrapping ensures that exceptions are handled properly.)